### ISSUE TITLE
Action onButtonPress and icon background color

### ISSUE THREAD

**Issue by matheusyuri7** [Raja0sama/rn-foreground-service#52]
Hello,

I'm a huge fan of this library.

I would like to know if it is possible to run an action when the notification button is pressed. I'm trying:

ReactNativeForegroundService.start({
        id: 111,
        title: 'Fitness Challenge',
        message: '0.02 km / 3 kcal',
        importance: '2',
        icon: 'ic_notifications',
        color: '#9A1E15',
        button: true,
        buttonText: "Pause",
        buttonOnPress: console.log("corinthians")
      });

![WhatsApp Image 2023-01-03 at 10 52 47 AM](https://user-images.githubusercontent.com/41174454/210370768-a587cd9e-b9a9-44b9-940b-186a36f0d6ec.jpeg)

Last question: Is it possible to change the icon background color? I would like to change that green if possible

**Comment by Raja0sama** [comment]
Yes, I have just pushed event support for the buttonPress, You can make use `eventListener` to get all the events. 

code would be something like this 
```
const unsub = ReactNativeForegroundService.eventListener(callBack)
// unsub when needed.
unsub()
```

callback will have parameters.

**Comment by matheusyuri7** [comment]
hello @Raja0sama thanks for the quick response.

I didn't understand how exactly it works. 

Could you provide an example, please?

I tried the following:
```
ReactNativeForegroundService.start({
        id: 111,
        title: 'Fitness Challenge',
        message: '0.02 km / 3 kcal',
        importance: '2',
        icon: 'ic_notifications',
        color: '#9A1E15',
        button: true,
        buttonText: "Pause",
        buttonOnPress: callback
      });
```

Another question: Is it possible to change the icon background color? I would like to change that green if possible

**Comment by Raja0sama** [comment]
Make a separate feature request for colour of the icon, i will make a PR and push it soon. 

Mean while, you need to understand this, When you can create a foreground service. And when you exit the app and click on what ever it is on the notification, it reopens up the application. 

So you have to set up a event listerenr there to listen for the events. 

SO when you open starts up, you get subscribed to listener and receive pending events in the callback.

**Comment by matheusyuri7** [comment]
Nice, I get it now! Thanks again for your help!

**Comment by Pauligrinder** [comment]
Just FYI: I couldn't get the callback system to work, but I did manage to use it via the Android native onNewIntent function -> that gets my defined mainOnPress string, which I then pass into the app and it works just as any deep link :)
